plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs

```diff
@@ -165,15 +165,17 @@ evaluateScriptRestricting
     -> Script -- ^ The script to evaluate
     -> [Data] -- ^ The arguments to the script
     -> (LogOutput, Either EvaluationError ())
-evaluateScriptRestricting _verbose costParams budget p args = swap $ runWriter @LogOutput $ runExceptT $ do
+evaluateScriptRestricting verbose costParams budget p args = swap $ runWriter @LogOutput $ runExceptT $ do
     appliedTerm <- mkTermToEvaluate p args
     model <- liftEither $ mkCostModel costParams
 
-    let (res, _) = UPLC.runCek
-          (PLC.toBuiltinsRuntime PLC.defDefaultFunDyn model)
+    let (res, _, logs) = UPLC.runCek
+          (PLC.toBuiltinsRuntime () model)
           (PLC.Restricting $ PLC.ExRestrictingBudget budget)
-          appliedTerm
+          (verbose == Verbose)
+         appliedTerm
 
+    tell $ Prelude.map Text.pack logs
     liftEither $ first CekError $ void res
 
 -- | Evaluates a script, returning the budget that the script would need to evaluate successfully.
```

Comment:
I suppose that this is yet another distinct budgeting mode.  In `Counting` mode we currently get back not just the total budget used, but also the individual costs for machine steps and builtins, which are only relevant to building/calibrating/validatng the cost model.  It's not too important here since this is something that we're running off-chain and the extra cost of collecting stuff we don't need isn't that important.  If the budgeting infrastructure becomes more flexible it might be nice to have a separate mode for just counting overall costs.  I'll add a note to SCP-1669.

Reply:
Agreed.

Reply:
>  I'll add a note to SCP-1669.

Thanks!

Reply:
I'm not sure it matters so much, but yeah, we could have such a mode.